pkg/injector/pod_patch.go

```diff
@@ -271,6 +281,59 @@ func profilingEnabled(annotations map[string]string) bool {
 	}
 }
 
+func appendQuantityToResourceList(quantity string, resourceName v1.ResourceName, resourceList v1.ResourceList) (*v1.ResourceList, error) {
+	q, err := resource.ParseQuantity(quantity)
+	if err != nil {
+		return nil, err
+	}
+	resourceList[resourceName] = q
+	return &resourceList, nil
+}
+
+func getResourceRequirements(annotations map[string]string) (*v1.ResourceRequirements, error) {
+	r := v1.ResourceRequirements{
+		Limits:   v1.ResourceList{},
+		Requests: v1.ResourceList{},
+	}
+	cpuLimit, ok := annotations[daprCPULimitKey]
```

Comment:
Can we create array of ResourceKeys and iterate over all the resource keys ? So instead of 4 repeatable code blocks , will have only single block?

Reply:
Actually, I looked into it but I prefer it to be more explicit and not introduce logic to know the exact resource type inside the single block.